test/helpers/kubectl.go

```diff
@@ -1842,6 +1879,56 @@ func (kub *Kubectl) CiliumServicePreFlightCheck() error {
 	return nil
 }
 
+// DeployETCDOperator deploys the etcd-operator k8s descriptors into the cluster
+// pointer by kub.
+func (kub *Kubectl) DeployETCDOperator() error {
+	const etcdOperatorPath = "../examples/kubernetes/addons/etcd-operator/"
+	deployFile := func(filename string) error {
+		cmdRes := kub.Apply(GetFilePath(etcdOperatorPath + filename))
+		if !cmdRes.WasSuccessful() {
+			return fmt.Errorf("Unable to deploy descriptor of etcd-operator %s: %s", filename, cmdRes.OutputPrettyPrint())
+		}
+		return nil
+	}
+
+	kub.Exec(fmt.Sprintf("%s '%s'", GetFilePath(etcdOperatorPath+"tls/certs/gen-cert.sh"), "cluster.local"))
+	kub.Exec(GetFilePath(etcdOperatorPath + "tls/deploy-certs.sh"))
+
+	for _, manifest := range etcdDeploymentFiles {
+		err := deployFile(manifest)
+		if err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+// DeleteETCDOperator delete the etcd-operator from the cluster pointed by
+// kub.
+func (kub *Kubectl) DeleteETCDOperator() error {
+	const etcdOperatorPath = "../examples/kubernetes/addons/etcd-operator/"
+	deleteFile := func(filename string) error {
+		cmdRes := kub.Delete(GetFilePath(etcdOperatorPath + filename))
+		if !cmdRes.WasSuccessful() {
+			return fmt.Errorf("Unable to deploy descriptor of etcd-operator %s: %s", filename, cmdRes.OutputPrettyPrint())
+		}
+		return nil
+	}
+
+	kub.Exec(fmt.Sprintf("%s '%s'", GetFilePath(etcdOperatorPath+"tls/certs/gen-cert.sh"), "cluster.local"))
+	kub.Exec(GetFilePath(etcdOperatorPath + "tls/deploy-certs.sh"))
```

Comment:
Why we need to execute this on Delete? I'm not sure

Reply:
Bad copy-paste.